x/dispute/keeper/msg_server_propose_dispute.go

```diff
@@ -25,6 +25,10 @@ func (k msgServer) ProposeDispute(goCtx context.Context, msg *types.MsgProposeDi
 	if msg.Fee.Amount.LT(layer.OnePercent) {
 		return nil, types.ErrMinimumTRBrequired.Wrapf("fee %s doesn't meet minimum fee required", msg.Fee.Amount)
 	}
+	// return an error if the proposer attempts to create a dispute on themselves while paying from their bond
+	if msg.PayFromBond && sender.String() == msg.Creator {
```

Comment:
this looks like msg creator is being compared to msg creator, comparison should be msg creator to reporter (msg.Report.Reporter) and comparing strings probably better to use strings.Equal